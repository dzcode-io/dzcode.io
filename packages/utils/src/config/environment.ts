export const environments = ["development", "staging", "production"] as const;

export type Environment = typeof environments[number];
