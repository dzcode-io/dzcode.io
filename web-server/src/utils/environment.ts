import { environments } from "@dzcode.io/utils/dist/config/environment";

export const getEnv = () => {
  let stage = process.env.STAGE;
  if (!environments.includes(stage)) {
    console.log(`No STAGE provided, falling back to "development"`);
    stage = "development";
  }

  return stage;
};
