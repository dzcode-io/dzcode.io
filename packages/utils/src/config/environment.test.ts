import { environments } from "./environment";

it("should match snapshot", () => {
  expect(environments).toMatchSnapshot();
});
