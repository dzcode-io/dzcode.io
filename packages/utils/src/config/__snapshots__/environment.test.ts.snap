// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  "development",
  "staging",
  "production",
]
`;
